apps/common/src/main/scala/org/lfdecentralizedtrust/splice/environment/Node.scala

```diff
@@ -86,6 +82,29 @@ abstract class Node[State <: AutoCloseable & HasHealth, PreInitializeState](
         additionalCodes = Seq(Status.Code.PERMISSION_DENIED),
       )
     }
+
+  override protected def initializeNode(
+      ledgerClient: SpliceLedgerClient
+  )(implicit tc: TraceContext): Future[State] = for {
+    _ <- preInitializeBeforeLedgerConnection()
+    initConnection = appInitStepSync("Acquire ledger connection") {
+      ledgerClient.readOnlyConnection(
+        this.getClass.getSimpleName,
+        loggerFactory,
+      )
+    }
+    (preInitializeState, serviceParty) <-
+      if (waitForPartyBeforePreinitialize) {
```

Comment:
I'm not sure this is strictly necessary. But I'm nervous about the validator trying to modify the connection while the SV app is still initializing and I don't really see a big downside to that.